From 2a27a50963974c63296f84fe58f01e336fa77619 Mon Sep 17 00:00:00 2001
From: mio <miyopan@e.email>
Date: Wed, 18 Feb 2026 02:35:16 +0000
Subject: [PATCH] Fix type mismatch error in app/Main.hs
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix error return type in the `findPkgDesc` call.

```
app/Main.hs:62:22: error: [GHC-83865]
    • Couldn't match type ‘Distribution.Simple.Errors.CabalException’
                     with ‘[Char]’
      Expected: String
        Actual: Distribution.Simple.Errors.CabalException
    • In the first argument of ‘fail’, namely ‘err’
      In the expression: fail err
      In a \case alternative: Left err -> fail err
   |
62 |     Left err -> fail err
   |                      ^^^
```
---
 app/Main.hs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/app/Main.hs b/app/Main.hs
index 7de3e3a..5b924dc 100644
--- a/app/Main.hs
+++ b/app/Main.hs
@@ -14,6 +14,7 @@ import Distribution.PackageDescription.Parsec (readGenericPackageDescription)
 import Distribution.Simple.PackageDescription (readGenericPackageDescription)
 #endif
 import Distribution.PackageDescription.PrettyPrint (showGenericPackageDescription)
+import Distribution.Simple.Errors (exceptionMessage)
 import Distribution.Simple.Utils (findPackageDesc)
 import Distribution.Uusi.Core
 import Distribution.Uusi.Types
@@ -59,5 +60,5 @@ uusiCabal actions originPath = do
 findCabal :: IO FilePath
 findCabal =
   findPackageDesc "." >>= \case
-    Left err -> fail err
+    Left err -> fail (exceptionMessage err)
     Right x -> pure x
-- 
2.53.0

